client/jetpack-connect/plans.jsx

```diff
@@ -230,19 +224,31 @@ class Plans extends Component {
 		mc.bumpStat( 'calypso_jpc_plan_selection', cartItem.product_slug );
 
 		addItem( cartItem );
-		this.redirecting = true;
 		this.props.completeFlow();
 		page.redirect( checkoutPath );
 	};
 
 	render() {
-		const { interval, isRtlLayout, selectedSite, translate } = this.props;
+		const {
+			canPurchasePlans,
+			hasPlan,
+			flowType,
+			interval,
+			isAutomatedTransfer,
+			isRtlLayout,
+			notJetpack,
+			selectedPlan,
+			selectedSite,
+			translate,
+		} = this.props;
 
 		if (
-			this.redirecting ||
-			this.hasPreSelectedPlan() ||
-			! this.props.canPurchasePlans ||
-			this.props.hasPlan
+			this.isFlowTypePaid( flowType ) ||
```

Comment:
Tightening up the logic for cases that shouldn't render. Added a couple: `notJetpack`, `isAutomatedTransfer`. Maybe this should come out into a separate function?

Thought about adding some kind of placeholder component, but that may be visually noisy during the redirect.